src/assets/js/application.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-env browser */
+
+/* global ClipboardJS:false */
+
+import ClipboardJS from 'clipboard';
+
+const btnHtml = [
+'<div class="bd-clipboard">',
+  '<button type="button" class="btn-clipboard" title="Copy to clipboard">',
+    '<svg class="bi" width="1em" height="1em" fill="currentColor">',
+      '<use xlink:href="#clipboard"/>',
+    '</svg>',
+  '</button>',
+'</div>'].join('');
+
+document.querySelectorAll('div.highlight')
+  .forEach((element) => {
+    element.insertAdjacentHTML('beforebegin', btnHtml)
+  })
+
+const clipboard = new ClipboardJS('.btn-clipboard', {
+  target: trigger => trigger.parentNode.nextElementSibling,
+  text: trigger => trigger.parentNode.nextElementSibling.textContent.trimEnd()
+})
+
+clipboard.on('success', (event) => {
+  const iconFirstChild = event.trigger.querySelector('.bi').firstElementChild
+  const namespace = 'http://www.w3.org/1999/xlink'
+  const originalXhref = iconFirstChild.getAttributeNS(namespace, 'href')
+  const originalTitle = event.trigger.title
+
+  event.clearSelection()
+  iconFirstChild.setAttributeNS(namespace, 'href', originalXhref.replace('clipboard', 'check2'))
+  event.trigger.title = 'Copied!'
+
+  setTimeout(() => {
+    iconFirstChild.setAttributeNS(namespace, 'href', originalXhref)
+    event.trigger.title = originalTitle
+  }, 2000)
+})
+
+/*clipboard.on('error', () => {
```

Comment:
@julien-deramond This is commented out. I didn't have the time so I left it here. Feel free to push any suggestions :)

Reply:
I suppose that we don't want here to embed Popper + use tooltips as we do in Bootstrap doc.

So I'd try something like that (not really tested in depth) if I've understood correctly what you'd like to achieve:

```diff
diff --git a/src/assets/js/application.js b/src/assets/js/application.js
index f3c93167..48c33cc5 100644
--- a/src/assets/js/application.js
+++ b/src/assets/js/application.js
@@ -13,9 +13,14 @@ const btnHtml = [
   '</button>',
 '</div>'].join('');
 
+const copyErrorCallout = [
+  '<div id="copy-error-callout" class="d-none alert alert-danger" role="alert">',
+  '</div>'].join('');
+
 document.querySelectorAll('div.highlight')
   .forEach((element) => {
     element.insertAdjacentHTML('beforebegin', btnHtml)
+    element.insertAdjacentHTML('beforeend', copyErrorCallout)
   })
 
 const clipboard = new ClipboardJS('.btn-clipboard', {
@@ -24,6 +29,11 @@ const clipboard = new ClipboardJS('.btn-clipboard', {
 })
 
 clipboard.on('success', (event) => {
+  const errorElement = document.getElementById('copy-error-callout')
+  if (errorElement) {
+    errorElement.classList.add('d-none')
+  }
+
   const iconFirstChild = event.trigger.querySelector('.bi').firstElementChild
   const namespace = 'http://www.w3.org/1999/xlink'
   const originalXhref = iconFirstChild.getAttributeNS(namespace, 'href')
@@ -39,7 +49,7 @@ clipboard.on('success', (event) => {
   }, 2000)
 })
 
-/*clipboard.on('error', () => {
+clipboard.on('error', () => {
   const modifierKey = /mac/i.test(navigator.userAgent) ? '\u2318' : 'Ctrl-'
   const fallbackMsg = 'Press ' + modifierKey + 'C to copy'
   const errorElement = document.getElementById('copy-error-callout')
@@ -49,5 +59,5 @@ clipboard.on('success', (event) => {
   }
 
   errorElement.classList.remove('d-none')
-  errorElement.insertAdjacentHTML('afterbegin', fallbackMsg)
-})*/
+  errorElement.innerHTML = fallbackMsg
+})
diff --git a/src/assets/scss/_bootstrap.scss b/src/assets/scss/_bootstrap.scss
index 4b552824..ebd53d82 100644
--- a/src/assets/scss/_bootstrap.scss
+++ b/src/assets/scss/_bootstrap.scss
@@ -13,6 +13,7 @@
 @import "bootstrap/grid";
 @import "bootstrap/tables";
 // @import "bootstrap/forms";
+@import "bootstrap/alert";
 @import "bootstrap/buttons";
 @import "bootstrap/transitions";
 @import "bootstrap/dropdown";
```

Reply:
We actually can just include the bundle. Can you take care of it please? Otherwise I'll have a look later and backport the related code from the bootstrap repo.

Reply:
OK I'll include (I'll try tonight) the bundle and handle it like we already do in our main repo docs.

Reply:
Tried something in https://github.com/twbs/blog/pull/330/commits/a37fb0ffc6439a1d6ccc4f774aa47e9b9847cff1. I let you check that and tell me your thoughts.